### PR TITLE
Mathics -> Mathics3

### PR DESCRIPTION
**Pull request by rocky**
And some URL corrections

See also https://mathics-development-guide.readthedocs.io/en/mathics-mathics3/

### CODE REVIEW

File: docs/extending/code-overview/ast.rst [hunk 1]
```diff
@@ -45,7 +45,7 @@ identifier name in a programming language and the variable it
 represents.
 
 The class definitions for these are given in `mathics.core.parser.ast
-<https://github.com/mathics/Mathics/tree/master/mathics/core/parser.ast>`_.
+<https://github.com/Mathics3/mathics-core/tree/master/mathics/core/parser.ast>`_.
```
Comment (TiagoCavalcante) [inline docs/extending/code-overview/ast.rst]: This link is broken

File: docs/extending/code-overview/object-classes.rst [hunk 2]
```diff
@@ -6,8 +6,8 @@ The fundamental classes that functions are built up from are described
 below.
 
 Most of these classes are defined in `mathics.builtin.base
-<https://github.com/mathics/Mathics/tree/master/mathics/builtin/base.py>`_
-or `mathics.core.expression <https://github.com/mathics/Mathics/tree/master/mathics/core/expression.py>`_.
+<https://github.com/Mathics3/mathics-core/tree/master/mathics/builtin/base.py>`_
+or `mathics.core.expression <https://github.com/mathics/Mathics3/tree/master/mathics-core/core/expression.py>`_.
```
Comment (TiagoCavalcante) [inline docs/extending/code-overview/object-classes.rst]: This link is broken

File: docs/extending/code-overview/object-classes.rst [hunk 3]
```diff
@@ -300,18 +300,18 @@ SympyConstant, MPMathConstant, and NumpyConstant
 SympyFunction and MPMathFunction
 ================================
 
-Which Class should be used for a Mathics Object?
-================================================
+Which Class should be used for a Mathics3 Object?
```
Comment (TiagoCavalcante) [inline docs/extending/code-overview/object-classes.rst]: Maybe "Class" -> "class"?

File: docs/extending/developing-code/calling-mathics-library-from-within-python-code.rst [hunk 4]
```diff
@@ -94,16 +94,16 @@ Although we set ``add_builtin`` explicitly for pedagodical purpose,
 leave it off in future examples.
 
 
-Mathics Results as Python Objects
----------------------------------
+Mathics3 Results as Python Objects
+----------------------------------
```
Comment (TiagoCavalcante) [inline docs/extending/developing-code/calling-mathics-library-from-within-python-code.rst:98]: Is there a reason for using `------` here? (other places use `=====`)

File: docs/extending/developing-code/contributing-developing.rst [hunk 5]
```diff
@@ -4,9 +4,9 @@ Contributing & Developing
 Contributing
 ------------
 
-Here is the basic workflow for contributions to the Mathics source code:
+Here is the basic workflow for contributions to the Mathics3 source code:
 
-1. Create your own github account and fork the `Mathics github
+1. Create your own github account and fork the `Mathics3 github
    repository <https://github.com/mathics/Mathics>`__ (see `github help
```
Comment (TiagoCavalcante) [inline docs/extending/developing-code/contributing-developing.rst]: This link have to be updated

File: docs/extending/developing-code/extending/case-studies/Undefined/checking.rst [hunk 6]
```diff
@@ -45,7 +45,7 @@ called ``mathics`` we can also see this builtin
 
     $ mathics --full-form
 
-    Mathics 5.0.3dev0
+xo    Mathics3 5.0.3dev0
```
Comment (TiagoCavalcante) [inline docs/extending/developing-code/extending/case-studies/Undefined/checking.rst]: I guess `xo` should be removed

File: docs/extending/developing-code/extending/tutorial/0-predefined.rst [hunk 7]
```diff
@@ -15,31 +15,31 @@ Add the above at the end to a file in `mathics.builtin
 <https://github.com/mathics/Mathics/tree/master/mathics/builtin.ast>`_
```
Comment (TiagoCavalcante) [inline docs/extending/developing-code/extending/tutorial/0-predefined.rst]: This link should be updated

File: docs/extending/packages.rst [hunk 8]
```diff
@@ -1,26 +1,26 @@
...
 
-If you are familiar with the Wolfram Language, a natural thing to do is to write in the Mathics dialect of the Wolfram Language.
+If you are familiar with the Wolfram Language, a natural thing to do is to write in the Mathics3 dialect of the Wolfram Language.
 
 The Wolfram Language ``Get`` or ``<<`` can be used to read in a file and evaluate each expression returning the last one.
 
 Higher-level packaging mechanism can be built from this.
 
-We have a very crude one for packages that come distributed with the Mathics core, and they can be found in `mathics/packages <https://github.com/mathics/Mathics/tree/master/mathics/packages>`_.
+We have a very crude one for packages that come distributed with the Mathics3 core, and they can be found in `mathics/packages <https://github.com/mathics/Mathics/tree/master/mathics/packages>`_.
```
Comment (TiagoCavalcante) [inline docs/extending/packages.rst]: This link should be updated

**Review by TiagoCavalcante — APPROVED** [review]
LGTM, I think that other links have to be updated too though

File: docs/extending/developing-code/calling-mathics-library-from-within-python-code.rst (hunk 4 above)
Comment (rocky) [reply docs/extending/developing-code/calling-mathics-library-from-within-python-code.rst:98]: The `-` and `=` are different section levels

**Pull request merged**